src/main/java/com/puppycrawl/tools/checkstyle/JavadocDetailNodeParser.java

```diff
@@ -559,6 +561,63 @@ private static String convertUpperCamelToUpperUnderscore(String text) {
         return result.toString();
     }
 
+    public DetailAST parse(DetailAST javadocCommentAst) {
+        blockCommentLineNumber = javadocCommentAst.getLineNo();
+
+        final String javadocComment = JavadocUtil.getJavadocCommentContent(javadocCommentAst);
+
+        // Use a new error listener each time to be able to use
+        // one check instance for multiple files to be checked
+        // without getting side effects.
+        final DescriptiveErrorListener errorListener = new DescriptiveErrorListener();
+
+        // Log messages should have line number in scope of file,
+        // not in scope of Javadoc comment.
+        // Offset is line number of beginning of Javadoc comment.
+        errorListener.setOffset(javadocCommentAst.getLineNo() - 1);
+
+        DetailAST tree = null;
+
+        try {
+            final JavadocCommentsLexer lexer =
+                        new JavadocCommentsLexer(CharStreams.fromString(javadocComment));
+
+            final CommonTokenStream tokens = new CommonTokenStream(lexer);
+
+            final JavadocCommentsParser parser = new JavadocCommentsParser(tokens);
+
+            // set prediction mode to SLL to speed up parsing
+            parser.getInterpreter().setPredictionMode(PredictionMode.SLL);
+
+            // remove default error listeners
+            parser.removeErrorListeners();
+
+            final JavadocCommentsParser.JavadocContext javadoc = parser.javadoc();
+
+            tree = new JavadocCommentsAstVisitor().visit(javadoc);
+
+            adjustFirstLineToJavadocIndent(tree,
+                        javadocCommentAst.getColumnNo()
+                                + JAVADOC_START.length());
+        }
+        catch (ParseCancellationException | IllegalArgumentException ex) {
+
+        }
+
+        return tree;
+    }
+
+    private void adjustFirstLineToJavadocIndent(DetailAST tree, int javadocColumnNumber) {
+        if (tree.getLineNo() == blockCommentLineNumber) {
+            ((DetailAstImpl) tree).setColumnNo(tree.getColumnNo() + javadocColumnNumber);
+            while (tree.hasChildren()) {
+                tree = tree.getFirstChild();
+                adjustFirstLineToJavadocIndent(tree, javadocColumnNumber);
```

Comment:
If this were to move forward, it may make more sense to try to move this into `JavadocCommentsAstVisitor` so it is done on node creation.